File: testdata/fixtures/go_project/service.go
package project

import "fmt"

// UserService handles user business logic.
type UserService struct {
	repo Repository
}

// NewUserService creates a new UserService.
func NewUserService(repo Repository) *UserService {
	return &UserService{repo: repo}
}

// GetUser retrieves a user by ID.
func (s *UserService) GetUser(id int) (*User, error) {
	user, err := s.repo.FindByID(id)
	if err != nil {
		return nil, fmt.Errorf("get user: %w", err)
	}
	return user, nil
}

// CreateUser creates a new user.
func (s *UserService) CreateUser(name, email string) (*User, error) {
	user := newUser(name, email)
	if err := s.repo.Save(user); err != nil {
		return nil, fmt.Errorf("create user: %w", err)
	}
	return user, nil
}

File: testdata/fixtures/py_project/service.py
from .models import User, create_user


class UserService:
    """Handles user business logic."""

    def __init__(self) -> None:
        self._users: list[User] = []

    def get_user(self, user_id: int) -> User | None:
        for user in self._users:
            if user.id == user_id:
                return user
        return None

    def create(self, name: str, email: str) -> User:
        user = create_user(name, email)
        self._users.append(user)
        return user
